File: billing_ui_app/src/app/core/models/billing/claim-details-model.ts
import { AuthorizationDiagnosisCode } from "../clients/authorization";
import { BillingProviderRequest } from "./claim-billing-provider";

export interface ClaimDetailsModel extends ClaimUpdateDetailsModel, ClaimNoteDetailsModel{
    dos: string;
    billingCode: string;
    diagnosis: string;
    units: number;
    billedAmount: number;
    expectedAmount: number;
    hours: number;
    paymentAmount: number;
    patientAmount: number;
    balanceAmount: number;
    adjustmentAmount: number;
    totalCount: number;
    associatedAppointmentsCount: number;
    reasonCode: string;
    renderingProvider: string;
    renderingProviderId: number;

    renderingProviderName?: string;
}

export interface ClaimUpdateDetailsModel extends ClaimUpdateModifiersModel{
    claimId: number;
    units: number;
    perUnitsCharge: number;
    billingCodeId?: number;
    diagnosis?: string;
    unitTypeId?: number;
    DateOfService?: Date;
    billingCode?: string;
    hours?: number;
    billedAmount?: number;
    expectedAmount?: number;
    paymentAmount?: number;
    patientAmount?: number;
    balanceAmount?: number;
    adjustmentAmount?: number;
    isSecondaryCode?: boolean;      
    pairBillingCodeId?: number;
    individualDateOfService?: any;
    renderingProviderId?: number;
}

export interface ClaimUpdateModifiersModel{
    id: number;
    modifier1: string;
    includeOnClaimMod1: boolean;
    modifier2: string;
    includeOnClaimMod2: boolean;
    modifier3: string;
    includeOnClaimMod3: boolean;
    modifier4: string;
    includeOnClaimMod4: boolean;
}

export interface ClaimStatusUpdateModel{
    claimId: number;
    claimStatusId: number;
}

export interface ClaimUpdateModifiersViewModel extends ClaimUpdateModifiersModel {
    isValid: boolean;
}

export interface ClaimNoteDetailsModel extends ClaimNoteAddModel{
    noteCreatorName: string | null;
    noteCreatedDate: Date | null;
}

export interface ClaimNoteAddModel{
    chargeId: number;
    noteText: string;
}

export interface ClaimUpdateDetailsModels{
    billingClaimDetailsModels: ClaimUpdateDetailsModel[];
    memberId: number;
}


export interface ClaimDetailsInfoModel extends ClaimDetailsInfoUpdateModel {
    id: number;
    patientId: number;
    patientName: string;
    funderId: number;
    funderName: string;
    funderTypeId: number;
    responsibleParty: string;
    dateOfServiceStart: Date;
    dateOfServiceEnd: Date;
    authorizationNumber: string;
    authorizationStatus?: string;

    referringProviderRequiredOnClaim: boolean;
    
    billedAmount: number;
    balanceAmount: number;
    paymentAmount: number;
    patientResponsibilityAmount: number;
   
    providerSignature: string;
    submissionCode: string;
    
    //from update interface
    claimId: number;
    diagnosisCodes: AuthorizationDiagnosisCode[];

    placeOfService: string;

    renderingProviderId: number;
    referringProviderId: number;
    billingProviderId: number;
    serviceFacilityId: number;

    patientReleaseAgreement: string;
    authorizePayment: string;
    benefitsAssignment: number;
    submissionReason: string;

    billingProviderOptionId: number;

    authorizationDetails: AuthorizationDetailsModel;
    serviceLineId: number;
    serviceLine:string;
    serviceId: number;
    primaryFunderId?: number;
    secondaryFunderId?: number;
}

export interface ClaimDetailsInfoUpdateModel{
    claimId: number;
    diagnosisCodes: AuthorizationDiagnosisCode[];

    renderingProviderId: number;
    referringProviderId: number;
    billingProviderId?: number;
    serviceFacilityId: number;

    renderingProvider: string;
    referringProvider: string;
    billingProvider: string;
    serviceFacility: string;
    placeOfService: string;

    placeOfServiceId: number;
    patientReleaseAgreementId: number;
    authorizePaymentId: number;
    submissionReasonId: number;
    benefitAssignmentId: number;
    originalClaim: string;
    note: string;  
    
}

interface AuthorizationDetailsModel {
    renderingProviderId?: number;
}


export class ClaimUpdateModel {
    isClaimUpdated: boolean = false;
    isChargeEntryUpdated: boolean = false;
    claimModel: ClaimDetailsInfoUpdateModel;
    chargeEntryModel: ClaimUpdateDetailsModels;
    impersonationUserName: string | null;
    billingProviderRequest?: BillingProviderRequest | null;
}

File: Microservices/BillingService/BillingService.Domain/Models/ClaimDetailsModel.cs
using Rethink.Services.Common.Enums.BH;
using Rethink.Services.Common.Enums.Billing;
using System;
using System.Collections.Generic;

namespace BillingService.Domain.Models
{
    public class ClaimDetailsModel
    {
        public int Id { get; set; }
        public string ClaimIdentifier { get; set; }
        public int? PatientId { get; set; }
        public string PatientName { get; set; }
        public string ResponsibleParty { get; set; }
        public DateTime DateOfServiceStart { get; set; }
        public DateTime DateOfServiceEnd { get; set; }
        public List<ClaimDiagnosisCodeModel> DiagnosisCodes { get; set; }
        public string AuthorizationNumber { get; set; }
        public string? AuthorizationStatus { get; set; } = "Yes";
        public string RenderingProvider { get; set; }
        public string ReferringProvider { get; set; }
        public string BillingProvider { get; set; }
        public string ServiceFacility { get; set; }
        public int? RenderingProviderId { get; set; }
        public int? ReferringProviderId { get; set; }
        public bool ReferringProviderRequiredOnClaim { get; set; }
        public int? BillingProviderId { get; set; }
        public int? ServiceFacilityId { get; set; }

        public decimal BilledAmount { get; set; }
        public decimal BalanceAmount { get; set; }
        public decimal PaymentAmount { get; set; }
        public decimal PatientResponsibilityAmount { get; set; }

        public string PlaceOfService { get; set; }
        public int PlaceOfServiceId { get; set; }
        public int? PatientReleaseAgreement { get; set; }
        public int? AuthorizePayment { get; set; }
        public int? BenefitsAssignment { get; set; }
        public string ProviderSignature { get; set; }
        public int SubmissionReason { get; set; }
        public string SubmissionCode { get; set; }
        public string OriginalClaim { get; set; }
        public string Note { get; set; }
        public ClaimStatus ClaimStatus { get; set; }
        public int? FunderId { get; set; }
        public string FunderName { get; set; }
        public FunderType? FunderTypeId { get; set; }
        public BillingProviderOptionType? BillingProviderOptionId { get; set; }

        public AuthorizationDetailsModel AuthorizationDetails { get; set; }
        public int? ServiceLineId { get; set; }

        public string? ServiceLine { get; set; }
        public int ServiceId { get; set; }
        public int PrimaryFunderId { get; set; }
        public int? SecondaryFunderId { get; set; }
    }


    public class AuthorizationDetailsModel
    {
        public int? RenderingProviderId { get; set; }
    }
}